resources/services/fsx/filesystems.go

```diff
@@ -0,0 +1,563 @@
+package fsx
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/fsx"
+	"github.com/aws/aws-sdk-go-v2/service/fsx/types"
+	"github.com/cloudquery/cq-provider-aws/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/diag"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+//go:generate cq-gen --resource filesystems --config filesystems.hcl --output .
+func Filesystems() *schema.Table {
+	return &schema.Table{
+		Name:         "aws_fsx_filesystems",
+		Description:  "A description of a specific Amazon FSx file system.",
+		Resolver:     fetchFsxFilesystems,
+		Multiplex:    client.ServiceAccountRegionMultiplexer("glue"),
+		IgnoreError:  client.IgnoreAccessDeniedServiceDisabled,
+		DeleteFilter: client.DeleteAccountRegionFilter,
+		Options:      schema.TableCreationOptions{PrimaryKeys: []string{"arn"}},
+		Columns: []schema.Column{
+			{
+				Name:        "account_id",
+				Description: "The AWS Account ID of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSAccount,
+			},
+			{
+				Name:        "region",
+				Description: "The AWS Region of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSRegion,
+			},
+			{
+				Name:        "creation_time",
+				Description: "The time that the file system was created, in seconds (since 1970-01-01T00:00:00Z), also known as Unix time.",
+				Type:        schema.TypeTimestamp,
+			},
+			{
+				Name:        "dns_name",
+				Description: "The Domain Name System (DNS) name for the file system.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("DNSName"),
+			},
+			{
+				Name:        "failure_details_message",
+				Description: "A message describing any failures that occurred during file system creation.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("FailureDetails.Message"),
+			},
+			{
+				Name:        "file_system_id",
+				Description: "The system-generated, unique 17-digit ID of the file system.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "file_system_type",
+				Description: "The type of Amazon FSx file system, which can be LUSTRE, WINDOWS, ONTAP, or OPENZFS.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "file_system_type_version",
+				Description: "The Lustre version of the Amazon FSx for Lustre file system, either 2.10 or 2.12.",
+				Type:        schema.TypeString,
+			},
```

Comment:
maybe remove file_system prefix in file_system table?

Reply:
Done